whitepaper/whitepaper.md

```diff
@@ -518,51 +518,53 @@ gets replaced by a traditional bank account holding USD.
-
+| Use Case Attribute Name    | Use Case Attribute Value |
+| -------------------------- | -------------------------------------------- |
+| Use Case Title             | Blockchain Migration |
+| Use Case                   | 1. `User A` runs an application/service on a source blockchain that is either permissioned or permissionless.<br>2. `User A` decides to use another blockchain instance (permissioned or permissionless) to run his/her application. <br>3. `User A` migrates the existing assets and data to the target blockchain.<br>4. `User A` runs the application/service on the target blockchain.
+| Interworking patterns      | Value transfer, Data transfer |
+| Type of Social Interaction | Asset and Data Transfer |
+| Narrative                  | `User A` who's application/service is operating on a source blockchain (e.g., a Hyperledger Fabric instance), would want to migrate the assets, data, and functionality to a target blockchain (e.g., Hyperledger Besu) to expand their reach. However, such migration requires lots of technical effort and time. The `Blockchain Migration` feature from `Cactus` can provide support for doing so, by connecting to the source and target blockchains, and performing the migration task.
+| Actors                     | 1. `User A`: The entity operating on the source blockchain which aims at performing a migration to the target blockchain. |
+| Goals of Actors            | 1. `User A`: Wants to be able to operate his/her application on the target blockchain while gaining its benefits such as better performance, lower cost, or enhanced privacy. |
+| Success Scenario           | Asset and data are available on the target blockchain enabling `User A`'s application to operate on the target blockchain.|
+| Success Criteria           | Migration starts at a set time, and all desired assets and data, as well as their histroy have been migrated in a decentralized way. |
+| Failure Criteria           | All desired assets and data, as well as their histroy not on the target blockchain. `User A` is unable migrate to target blockchain for a number of reasons, such as inability to recreate the same smart contract logic, inability to arbitary recreate native assets on target blockchain, and lack of access to private keys of external accounts. It may also be cost and time prohibitive to recreate entire history on the target blockchain. |
```

Comment:
I think proposed text is too much.
I think any side effects should not be occurred on the failure considering the nature of transition.
Because transition is one-way operation, retrying the operation should be considered instead.

Reply:
Does too much text refer to "Failure Criteria"? While retrying will mostly solve a problem, blockchain migration likely require a single-shot process rather than flushing target database if something goes wrong. Anyway, would the concern be addressed if I remove "It may also be cost and time prohibitive ..." sentence?

Reply:
No. I meant if CACTUS application will try to recover the failure it should be described, but this particular use case should not be fail unless it was rejected by participants.
More specially,  what can make it fail in your use case ? I believe that the transition should not be never rejected if system is correctly configured. And if it should be rejected there are no need to record the fact of the operation on both ledgers.

Reply:
Retry is needed and likely to be essential in setting multiple assets/data and transactions like in any distributed system. Hence, not mentioned explicitly. However, recording of transactions replayed on the target blockchain can lead to rejection or result in a state that's inconsistent with the state on source blockchain. This may not be fixed by retrying. I added "... e.g., when transactions replayed on target blockchain are reordered". 
Other cases are inspired from Failure Criteria in use case 3.4 and 3.7 and focus on under what conditions migration can be done. Happy to remove these if they don't fit.

Reply:
IMHO, history of record cannot be transferred unless all participants of the sourced ledger are agreed to migrate the ledger hosting system. But think again, why the migration process is needed for this case.
That is an implementation issue , how old data format can be converted with new one, which is not interoperability.
I suggest you focusing on asset level migration only.

Reply:
Remove all discussion related to history in multiple places.
Anyway, about this topic - I was thinking of recreating the same history by replaying the entire transaction trace (if someone really really need it). That way, any data format incompatibility issues can be addressed too. Even in our paper, we don't recommend this option due to cost, time, and even impracticality. If source blockchain is decommissioned after migration, having a copy of source blockchain history as a proof of exist entry is the easiest.